### PR TITLE
Fix right click not working on non player npcs

### PR DESCRIPTION
**Pull request by alexcastro1919**
As the title says, now non player npcs support right click.

### CODE REVIEW

**Comment by OliverSchlueter** [comment]
Could you describe under which conditions right clicking non player npcs is not working?
I've tried a few types and it worked for me.

**Comment by alexcastro1919** [comment]
I tried with CHEST_BOAT and WANDERING_TRADER and none worked

**Comment by alexcastro1919** [comment]
This was the condition causing the issue
`if (!event.isAttack() && (event.getHand() == EquipmentSlot.HAND || event.getClickedRelativePosition() != null))`
I suppose it was because getClickedRelativePosition returned not null (and obviously right click is not an attack).

File: src/main/java/de/oliver/fancynpcs/listeners/PlayerUseUnknownEntityListener.java [hunk 1]
```diff
@@ -8,15 +8,28 @@
...
+
+        if (event.getHand() != EquipmentSlot.HAND) return;
+
         Npc npc = FancyNpcs.getInstance().getNpcManagerImpl().getNpc(event.getEntityId());
-        if (npc == null) {
-            return;
-        }
+        if (npc == null) return;
 
+        /* This is not needed
```
Comment (OliverSchlueter) [inline src/main/java/de/oliver/fancynpcs/listeners/PlayerUseUnknownEntityListener.java:32]: You can actually remove this, if it's not needed.

**Review by OliverSchlueter — CHANGES_REQUESTED** [review]

**Review by alexcastro1919 — COMMENTED** [review]
Removed

**Review by OliverSchlueter — APPROVED** [review]

**Comment by OliverSchlueter** [comment]
Thank you!

**Comment by OliverSchlueter** [comment]
Not working on Folia
image

I reverted the commit for now, will need to investigate the actual problem

**Pull request merged**